src/index.ts

```diff
@@ -232,12 +238,14 @@ export class WaxJS {
     if (event.data.type === "TX_SIGNED") {
       const { verified, signatures, whitelistedContracts } = event.data;
       if (!verified || signatures == null) {
+        this.waxEventSource.closeIframe();
```

Comment:
Looks like this `closeIframe` logic is required for this whole block so might as well put it above this if statement and remove the one above `return signatures` at line 245